app/policies/rent_policy.rb

```diff
@@ -0,0 +1,13 @@
+class RentPolicy < ApplicationPolicy
+  attr_reader :user, :rent
+
+  def initialize(user, rent)
+    @user = user
+    @rent = rent
+  end
+
+  # User must only index yours rents
+  def index?
+    user.id.present?
```

Comment:
As I said before, it's nearly impossible to use `authorize` with an index endpoint, but if we would do it, here we should check that all the records belongs to the logged in user. This could be a solution to achieve that

```ruby
def index?
  # the variable name is "rent" but it contains a collection
  rent.to_a.all? { |record| record.user_id == user.id }
end
```

Of course this isn't good at all because you iterate over all the Rents of the app, and if any of them belongs to another user this will return false and will break your index endpoint.